pkg/providers/provider.go

```diff
@@ -66,3 +67,8 @@ type MachineConfig interface {
 	GetNamespace() string
 	GetName() string
 }
+
+// KubeconfigReader reads the kubeconfig secret on a cluster and copies the contents to a writer.
+type KubeconfigReader interface {
+	WriteKubeconfig(ctx context.Context, clusterName, kubeconfig string, w io.Writer) error
```

Comment:
just realized the struct is named reader but its only method is a write

Reply:
renamed to KubeconfigWriter